liby2/src/Y2SerialComponent.cc

```diff
@@ -344,23 +344,25 @@ bool Y2SerialComponent::initializeConnection()
 
 	 // first write one space so the other side gets
 	 // fullfilled its necessities
-	 write(fd_serial, space_buf, 1);
+	 (void) (write(fd_serial, space_buf, 1) == 1);
 
 	 // now wait for a space to be read
 	 // (the other side has the same behaviour)
 	 if (await_readable(TIMEOUT))
 	 {
-	    read(fd_serial, buf, 1);	            // read one byte
+	    size_t r = read(fd_serial, buf, 1);	    // read one byte
 
-	    if (buf[0] == ' ') spaces_read++;       // was a space
-	    else               spaces_read = 0;     // trash read --> reset
+	    if (r == 1 && buf[0] == ' ')
+                spaces_read++;                      // was a space
+	    else
+                spaces_read = 0;                    // trash read --> reset
 	 }
       }
 
       // now that we've got our NUMSPACES spaces write 2 * NUMSPACES spaces to
       // the other side to avoid syncing problems. the remote parser will kindly
       // ignore them.
-      write(fd_serial, space_buf, 2 * NUMSPACES);
+      (void) (write(fd_serial, space_buf, 2 * NUMSPACES) == 2 * NUMSPACES);
```

Comment:
so if it fail we can have sync problems and do not know reasons? I think we should at least log it